framework/auth/core.py

```diff
@@ -164,10 +165,18 @@ def logged_in(self):
     def from_kwargs(cls, request_args, kwargs):
```

Comment:
And it's weird to me to have class constructor called `from_kwargs`. Why not just use `__init__`?